activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,17 @@
+*   Fix ActiveRecord::Relation #one? and #many? when query contains a .group()
+
+    These methods rely on #size, which sometimes performs a #count, but #count
```

Comment:
I think it would be better to talk about how things will work going forward, rather than re-stating the details of what was broken.

So how about something like this for the changelog body:

------------------

If a query contains a `group()`, `#one?` and `#many?` are now determined by the number of records that were grouped by.

For example, if there is a single post:

```ruby
Post.joins(:comments).group('posts.id').one? # true
Post.joins(:comments).group('posts.id').many? # false
```

If there's multiple posts:

```ruby
Post.joins(:comments).group('posts.id').one? # false
Post.joins(:comments).group('posts.id').many? # true
```

See #41870 for more examples.

Reply:
Thanks, good shout. I've modified the changelog to state the new behavior, rather than repeat the issue details.